### PR TITLE
Import during Welcome

### PR DESCRIPTION
**Pull request by bsclifton**
Tracked with https://github.com/brave/brave-browser/issues/1530

Code is rough, but this does work (ex: you can pick an option and import will happen)

TODOs:
- [x] finalize how actual screen should look / work (cc: @rebron / @rossmoody)
- [x] fix the UI 
- [x] remove `Safari` since it doesn't work (see https://github.com/brave/brave-browser/issues/2710)
- [x] remove `Bookmarks HTML File` option (since it requires an open file dialog)
- [x] import button should be grayed out unless user picks an option

screenshots from proof of concept
image

image

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
- Open go to the second screen of the welcome page
- Button should be disabled with the default option as "import from.." in the drop down
- 'Safari' and 'Bookmarks HTML file' should not be options in the drop down
- Selecting a valid option will enable the "Import" button
- Clicking the import button will import all setting from the source you import from
- Clicking the import button will send you to the next page

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by bsclifton** [comment]
Heads up @imptrx - since you're working on https://github.com/brave/brave-core/pull/2479 😄 
cc: @cezaraugusto 

I'll likely need some front-end assistance here (once we have proper mock-ups)

**Comment by rossmoody** [comment]
Here is a revised visual direction for this: 

Welcome Light

Chatted with @rebron, had a few thoughts but otherwise good to go. 
1. Seems smart to remove Safari for now, as the fix looks [very touchy](https://microsoftedgesupport.microsoft.com/hc/en-us/articles/360022521994-How-to-import-browser-data-from-Safari-in-Mac-OS-Mojave-10-14-4)
2. The goal of this panel is an easy-mode import process (for non advanced users) so it would be best to just import whatever is available for any given option in the dropdown without the additional checkbox configurations. Same rationale for omitting Bookmark HTML option. We are referencing the advanced import option in the copy now. 
3. We have a pattern of prompting the user to make a selection with the disabled button aesthetic and since we can't assume the browser they normally use, we should stick to that convention for now

### Copy
```
Set up your usual browsing environment by importing bookmarks and settings from your old browser. Advanced import options can be found in Settings.
```

**Comment by rossmoody** [comment]
@bsclifton One more question, Raf mentioned a scenario where a user might actually not be able to import anything. What does this scenario look like? And would it be possible to omit this screen entirely in that instance?

**Comment by bsclifton** [comment]
@rossmoody I think Safari would be the only case - in that case, we can remove Safari from the drop down.  If there are no items from the drop down, we could definitely skip the screen

File: components/brave_welcome_ui/components/screens/importBox.tsx [hunk 1]
```diff
@@ -5,23 +5,58 @@
...
+    }
+  }
+
+  onChangeImportSource = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    if (event.target.value === '') {
+      this.setState({ selectedBrowserProfile: null })
+      return
+    }
+
+    let selectedEntry = this.props.browserProfiles.find((entry: Welcome.BrowserProfile) => {
```
Comment (cezaraugusto) [inline components/brave_welcome_ui/components/screens/importBox.tsx]: wdyt about moving out this check to a helper file?

File: components/brave_welcome_ui/components/screens/importBox.tsx [hunk 2]
```diff
@@ -5,23 +5,58 @@
...
+
+    if (selectedEntry) {
+      this.setState({ selectedBrowserProfile: selectedEntry })
+    }
+  }
+
+  onHandleImport = () => {
+    const { onClick } = this.props
+    let sourceBrowserProfileIndex: number
+    sourceBrowserProfileIndex = this.state && this.state.selectedBrowserProfile && this.state.selectedBrowserProfile.index || 0
```
Comment (cezaraugusto) [inline components/brave_welcome_ui/components/screens/importBox.tsx]: same as above

File: components/brave_welcome_ui/components/screens/importBox.tsx [hunk 3]
```diff
@@ -32,13 +67,33 @@ export default class ImportBox extends React.PureComponent<Props, {}> {
...
-            text={getLocale('import')}
-            onClick={onClick}
-          />
+          <SelectGrid>
+            <SelectBox
+              onChange={this.onChangeImportSource}
+            >
+              <option key={0} value=''>{getLocale('importFrom')}</option>
+              {
+                (browserProfiles && Array.isArray(browserProfiles) && browserProfiles.length > 0)
```
Comment (cezaraugusto) [inline components/brave_welcome_ui/components/screens/importBox.tsx]: ditto above

File: components/brave_welcome_ui/constants/welcome_types.ts [hunk 4]
```diff
@@ -3,11 +3,10 @@
...
 export const enum types {
-  IMPORT_NOW_REQUESTED = '@@welcome/IMPORT_NOW_REQUESTED',
+  IMPORT_BROWSER_DATA_REQUESTED = '@@welcome/IMPORT_BROWSER_DATA_REQUESTED',
   GO_TO_TAB_REQUESTED = '@@welcome/GO_TO_TAB_REQUESTED',
   CLOSE_TAB_REQUESTED = '@@welcome/CLOSE_TAB_REQUESTED',
   CHANGE_DEFAULT_SEARCH_PROVIDER = '@@welcome/CHANGE_DEFAULT_SEARCH',
-  IMPORT_DEFAULT_SEARCH_PROVIDERS_STARTED = '@@welcome/IMPORT_DEFAULT_SEARCH_PROVIDERS_STARTED',
   IMPORT_DEFAULT_SEARCH_PROVIDERS_SUCCESS = '@@welcome/IMPORT_DEFAULT_SEARCH_PROVIDERS_SUCCESS',
-  IMPORT_DEFAULT_SEARCH_PROVIDERS_FAILURE = '@@welcome/IMPORT_DEFAULT_SEARCH_PROVIDERS_FAILURE'
+  IMPORT_BROWSER_PROFILES_SUCCESS= '@@welcome/IMPORT_BROWSER_PROFILES_SUCCESS'
```
Comment (cezaraugusto) [inline components/brave_welcome_ui/constants/welcome_types.ts]: nit: not sure why lint didn't spot this one but there should be a space after `IMPORT_BROWSER_PROFILES_SUCCESS`

File: components/brave_welcome_ui/welcomeUtils.ts [hunk 5]
```diff
@@ -16,3 +16,22 @@ export const getSearchEngineProviders = () => {
       })
   }
 }
+
+export const getValidBrowserProfiles = (browserProfiles: Array<Welcome.BrowserProfile>): Array<Welcome.BrowserProfile> => {
+  const result = browserProfiles.reduce((filteredProfiles, profile) =>
+    (profile.name === 'Safari' || profile.name === 'Bookmarks HTML File')
```
Comment (cezaraugusto) [inline components/brave_welcome_ui/welcomeUtils.ts]: sorry safari :P

**Comment by bsclifton** [comment]
🎉

**Pull request merged**